Redmine.pm

```diff
@@ -449,7 +480,7 @@ sub authz_handler {
 
 	my ($identifier, $project_id, $is_public, $status);
 
-	if($identifier = $cfg->{Project}) {
+	if($identifier = $cfg->{Project} and $cfg->{RepositoryType} ne 'None') {
```

Comment:
You should test `$cfg->{RepositoryType} eq 'None'` and handle it apart from the other cases instead of adding a condition to each tests. 

BTW: for boolean *expressions*, use `&&` not `and`.